### ISSUE TITLE
[vmauth]: TLS Errors Due To Bundled `prometheus-config-reloader` Image

### ISSUE THREAD

**Issue by VioletCranberry** [VictoriaMetrics/operator#755]
There are two preset variables:

```
VM_VMAGENTDEFAULT_CONFIGRELOADIMAGE: quay.io/prometheus-operator/prometheus-config-reloader:v0.58.0
VM_VMAUTHDEFAULT_CONFIGRELOADIMAGE:  quay.io/prometheus-operator/prometheus-config-reloader:v0.48.1
```

The second variable is specifically employed to reconcile the `config-reloader` sidecar container within the `vmauth` resource. The relevant function can be found [here](https://github.com/VictoriaMetrics/operator/blob/21f72037c5b0718b34de529e4329497283e46551/controllers/factory/vmauth.go#L439), and `c.VMAuthDefault.ConfigReloadImage` is the default variable used when there are no custom overrides.

Starting with version `0.55` of the `prometheus-config-reloader`, TLS certificate verification is skipped by default (refer to [this pull request](https://github.com/prometheus-operator/prometheus-operator/pull/4592)). This creates some confusion for a couple of reasons:

1. The Operator utilizes the same image, but tags are different when reconciling the init-container and the sidecar, which makes it complicated to understand.
2. Errors will be generated by the related function because the version `v0.48.1` will attempt to validate localhost, as demonstrated by the error logs provided:
```
│ level=error ts=2023-09-12T09:35:26.234847688Z caller=runutil.go:101 msg="function failed. Retrying in next tick" err="trigger reload: reload request failed: Post \"[https://localhost:443/-/reload](https://localhost/-/reload)\": x509: certificate is valid for test-service.test-namespace.svc, not localhost"                                                               
│ level=error ts=2023-09-12T09:35:31.234722253Z caller=runutil.go:101 msg="function failed. Retrying in next tick" err="trigger reload: reload request failed: Post \"[https://localhost:443/-/reload](https://localhost/-/reload)\": x509: certificate is valid for test-service.test-namespace.svc, not localhost"                                                              
│ level=error ts=2023-09-12T09:35:36.243438785Z caller=runutil.go:101 msg="function failed. Retrying in next tick" err="trigger reload: reload request failed: Post \"[https://localhost:443/-/reload](https://localhost/-/reload)\": x509: certificate is valid for test-service.test-namespace.svc, not localhost"  
```

In an ideal scenario, `VM_VMAUTHDEFAULT_CONFIGRELOADIMAGE` should also default to `quay.io/prometheus-operator/prometheus-config-reloader:v0.58.0`. This would also ensure that the built-in prometheus-config-reloader operates as expected out-of-the-box, eliminating the need for users to supply custom overrides to correct its default functionality.

**Comment by f41gh7** [comment]
I think, in future we have to move into our own config-reloader implementation. It significantly improves time for configuration sync between k8s objects and pod local configuration files. And it allows us to use a single variable for all config-reloader containers.

**Comment by Haleygo** [comment]
Should be addressed by https://github.com/VictoriaMetrics/operator/commit/7c1bd04bcc445091710d1b7b179a8995d4f7e748 and is available since [v0.39.0](https://github.com/VictoriaMetrics/operator/releases/tag/v0.39.0).
